src/basic/log.c

```diff
@@ -588,6 +592,20 @@ static int log_do_header(
         return 0;
 }
 
+static void log_do_context(struct iovec *iovec, size_t iovec_len, size_t *n) {
+        assert(iovec);
+        assert(n);
+
+        LIST_FOREACH(ll, c, _log_context)
+                STRV_FOREACH(s, c->fields) {
+                        if (*n >= iovec_len - 2)
```

Comment:
might underflow if iovec_len is smaller than 2?

add assert? or other way to handle corner case?